### PR TITLE
51 ci caching docker images in GitHub

### LINKED ISSUE #51: CI/ Caching docker images in github
The goal of this issue is to caching every docker images in ordre to make it faster for other builds : 

- [x] Write the global job
- [x] Caching according to the following command : docker-compose up --build

### LINKED ISSUE #52: Docker/take .env file into consideration when running docker-compose
## Why ?

When running `docker-compose up`, variables like ports or connection URI will not match the desired environment.

This is problematic since it may cause connection errors when services will try to connect to each other

## How to fix ?

- Load env file using docker-compose cli option `--env-file .prod.env | .preprod.env | .dev.env`, this would imply that all needed variables are put in a single .env file
- Specify build args to docker-compose and select corresponding .env file in each context

### LINKED ISSUE #70: Docker/Apollo building image

### PR DESCRIPTION
**Pull request by Bootoyka**
### Update CI

In this pr CI/CD takes it to a whole new level, and is now able to : 

- [x] CodeQL, to scan code while pushing on main
- [x] Lint, to scan and discover vulnerabilities in code
- [x] Service images are now cached in github
- [x] Jobs are splited then they can be analysed one by one

### Dev build of project
This pr delivers a docker-compose working with every dockerfile needed in the subject and now able to :

 - [x] build & run every services asked in the subject
 - [x] fully respects the subject
 - [x] Create database ready to fill in
 - [x] A start-docker.sh used to start all services or specific services to work on it
 - [x] A clean-docker.sh used to clean every unused images, containers, volumes & dangling resources

### Need to fix

- [ ] CI job to_github, multiple conditions success working properly

> The environment configuration needed to make it work properly is noted in the Readme.md added in this pr.
> Note that a passage is dedicated to the presentation of each service, so you are invited to fill in your part precisely, thanking you.
 @aurelenc @EmileEttel @Cavonstavant @majent4 
 

> Pls test you own part using these tools and notify if something doesn't work

#### Closing issues

close #51 
close #52 
close #70

### CODE REVIEW

**Review by Cavonstavant — APPROVED** [review]
:rocket:

File: docker-compose.yml [hunk 1]
```diff
@@ -23,18 +36,31 @@ services:
...
+      POSTGRES_DB: ${POSTGRES_DB}
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    networks:
+      - server_network
+    ports:
+      - "5432:5432"
+networks:
+  server_network:
+    driver: bridge
```
Comment (majent4) [inline docker-compose.yml:63]: Isn't bridge the default network driver reachable by each services ?
Did you mean to create a custom one ?

Reference :
https://docs.docker.com/compose/networking/#specify-custom-networks

**Review by majent4 — COMMENTED** [review]
docker compose suggestion

**Review by EmileEttel — APPROVED** [review]
LGTM

**Review by Cavonstavant — APPROVED** [review]
Please be aware that the string POSTGRES_URI needs also to be changed in the `schema.prisma` file in order to keep consistency

File: docker-compose.yml (hunk 1 above)
Comment (Bootoyka) [reply docker-compose.yml:63]: the goal was to network only containers supposed to communicate together, but after reading i should test if it's still working only by default, good one thx

Comment (majent4) [reply docker-compose.yml:63]: Ok, conversation resolved

**Comment by Bootoyka** [comment]
It's already done, right ?

**Review by majent4 — APPROVED** [review]

**Review by aurelenc — APPROVED** [review]

**Pull request merged**